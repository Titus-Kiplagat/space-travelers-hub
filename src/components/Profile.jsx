import { useSelector } from 'react-redux';
import { ListGroup } from 'react-bootstrap';
import RocketListItem from './RocketListItem';

const Profile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  return (
    <div className="px-5">
      <h2 className="font-bold">Reserved Rockets</h2>
      {reservedRockets.length === 0 && <p className="fs-2">You have no reserved rockets</p>}
      <ListGroup className="my-2">
        {reservedRockets.map((rocket) => (
          <RocketListItem key={rocket.id} rocket={rocket} />
        ))}
      </ListGroup>
    </div>
  );
};

export default Profile;
