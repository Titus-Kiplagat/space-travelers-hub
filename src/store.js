import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './redux/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});

export default store;
