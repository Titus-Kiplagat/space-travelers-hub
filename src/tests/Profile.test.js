import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../components/Profile';
import store from '../store';

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
