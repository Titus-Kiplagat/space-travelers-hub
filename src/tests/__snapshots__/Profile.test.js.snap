// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="px-5 d-flex justify-content-between gap-5"
>
  <div
    className="w-100"
  >
    <h3
      className="font-bold py-2"
    >
      My Missions
    </h3>
    <p
      className="fs-2"
    >
      You have no reserved rockets
    </p>
    <div
      className="my-2 list-group"
      onKeyDown={[Function]}
    />
  </div>
  <div
    className="w-100"
  >
    <h3
      className="font-bold py-2"
    >
      My Rockets
    </h3>
    <p
      className="fs-2"
    >
      You have no reserved rockets
    </p>
    <div
      className="my-2 list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
